web_src/js/components/RepoContributors.vue

```diff
@@ -0,0 +1,360 @@
+            />
+          </div>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import {SvgIcon} from '../svg.js';
+import {
+  Chart,
+  Title,
+  Tooltip,
+  Legend,
+  BarElement,
+  CategoryScale,
+  LinearScale,
+  TimeScale,
+  PointElement,
+  LineElement,
+  Filler,
+} from 'chart.js';
+import zoomPlugin from 'chartjs-plugin-zoom';
+import {Line as CLine} from 'vue-chartjs';
+import 'chartjs-adapter-dayjs-4/dist/chartjs-adapter-dayjs-4.esm';
+import $ from 'jquery';
+
+const {pageData} = window.config;
+
+Chart.register(
+  TimeScale,
+  CategoryScale,
+  LinearScale,
+  BarElement,
+  Title,
+  Tooltip,
+  Legend,
+  PointElement,
+  LineElement,
+  Filler,
+  zoomPlugin
+);
+
+export default {
+  components: {CLine, SvgIcon},
+  props: {
+    locale: {
+      type: Object,
+      required: true
+    },
+  },
+  data: () => {
+    return {
+      isLoading: false,
+      errorText: '',
+      totalStats: {},
+      sortedContributors: {},
+      repoLink: pageData.repoLink || [],
+      type: pageData.contributionType,
+      contributorsStats: [],
+      dateFrom: null,
+      dateUntil: null,
+      startDate: null,
+      endDate: null,
+    };
+  },
+  mounted() {
+    this.fetchGraphData();
+
+    $('.ui.dropdown').dropdown({
+      onChange: (val) => {
+        this.dateFrom = this.startDate;
+        this.dateUntil = this.endDate;
+        this.type = val;
+        this.sortContributors();
+      }
+    });
+  },
+  methods: {
+    sortContributors() {
+      const contributors = this.filterContributorWeeksByDateRange();
+      const sortingCriteria = `total_${this.type}`;
+      this.sortedContributors = Object.values(contributors).filter((contributor) => contributor[sortingCriteria] !== 0).sort((a, b) =>
+        a[sortingCriteria] > b[sortingCriteria] ? -1 : a[sortingCriteria] === b[sortingCriteria] ? 0 : 1
+      ).slice(0, 100);
+    },
+    async fetchGraphData() {
+      this.isLoading = true;
+      try {
+        const response = await fetch(`${this.repoLink}/activity/contributors/data`);
+        const data = await response.json();
+        const {total, ...rest} = data;
+        this.contributorsStats = rest;
+        this.dateFrom = new Date(total.weeks[0].week);
+        this.dateUntil = new Date();
+        this.startDate = this.dateFrom;
+        this.endDate = this.dateUntil;
+        this.sortContributors();
+        this.totalStats = total;
+        this.errorText = '';
+      } catch (e) {
+        this.errorText = e.message;
+      } finally {
+        this.isLoading = false;
+      }
+    },
+
+    filterContributorWeeksByDateRange() {
+      const filteredData = {};
+
+      const data = this.contributorsStats;
+      for (const key of Object.keys(data)) {
+        const user = data[key];
+        user['total_commits'] = 0;
+        user['total_additions'] = 0;
+        user['total_deletions'] = 0;
+        user['max_contribution_type'] = 0;
```

Comment:
No, it should be `max_contribution_type`. Here is why:

When user changes contribution type and zooms a little bit, I want to make sure all the contributor graphs share the same max Y value so that it is easier to compare them. This requires me to know each contributors' `max_contribution_type` within the selected time range so that I can compare them and set `max([all contributors' max_contribution_type])` as the maximum value for all graphs.

If contribution type is "deletions" and selected time range is Jan 1 - May 13, `max_contribution_type` would mean *maximum number of deletions between Jan 1 - May 13* for each contributor.